packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -172,13 +174,16 @@ abstract class TransitionRoute<T> extends OverlayRoute<T> {
   }
 
   @override
-  void install(OverlayEntry insertionPoint) {
+  void install() {
     assert(!_transitionCompleter.isCompleted, 'Cannot install a $runtimeType after disposing it.');
     _controller = createAnimationController();
     assert(_controller != null, '$runtimeType.createAnimationController() returned null.');
     _animation = createAnimation();
     assert(_animation != null, '$runtimeType.createAnimation() returned null.');
-    super.install(insertionPoint);
+    super.install();
+    if (_animation.status == AnimationStatus.completed && overlayEntries.isNotEmpty) {
```

Comment:
Why do we need this if? Isn't this happening in the _didPushOrReplace from didPush and didAdd?